### ISSUE TITLE
Board Review: Management Plane Namespace Review Workloads.SapVirtualInstance

### ISSUE THREAD

**Issue by adityashukla-msft** [Azure/azure-sdk#7077]
Thank you for submitting this review request. Thorough review of your management library namespaces ensures that your library names are consistent with the guidelines and the consumers of your management library have a consistently good experience when using Azure.

To ensure consistency, all language library names will generally be reviewed together.

**Before submitting, ensure you adjust the title of the issue appropriately.**

**Note that the required material must be included before a meeting can be scheduled.**

## Contacts and Timeline

* Responsible service team: Azure Workloads Platform as Service\Azure SAP Business Platform as Service (ACSS) - Engineering
* Main contacts: adityashukla@microsoft.com, ajgupt@microsoft.com, sushantyadav@microsoft.com, swbehl@microsoft.com, shayaksarkar@microsoft.com, svagadia@microsoft.com, chpadh@microsoft.com, gargankit@microsoft.com
* Expected code complete date: Already complete.
* Expected release date: January 2024

## About the Service (required)

* Link to documentation introducing/describing the service: https://learn.microsoft.com/en-us/azure/sap/center-sap-solutions/overview
* Link to the service REST APIs Specs: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json
* Link to GitHub issue for previous review sessions, if applicable:  

## Namespace Proposals (required per language)

**In the examples below please replace every occurrence of [ResourceProviderName] with your the service resource provider name. Be sure to keep the casing shown with [ResourceProviderName] when replacing it with the service resource provider name.**

- **.NET:** `Azure.ResourceManager.WorkloadsSapVirtualInstance`
- **Java:** `azure-resourcemanager-workloadssapvirtualinstance (com.azure.resourcemanager.workloadssapvirtualinstance)`
- **Go/Golang:** `sdk/resourcemanager/workloadssapvirtualinstance/armworkloadssapvirtualinstance`
- **JavaScript:** `@azure/arm-workloadssapvirtualinstance`
- **Python:** `azure-mgmt-workloadssapvirtualinstance`

## Thank you!

**Comment by ArthurMa1978** [comment]
Hey @adityashukla-msft, since the Workloads SDK is already in GA, splitting the SAP Virtual Instance from Workloads into a new library will cause a breaking change. Therefore, please conduct the breaking change review prior to the name space review.

**Comment by ajaygit158** [comment]
Hi @ArthurMa1978 

Could you explain why this would be a breaking change?
There is no workloads SDK for this API version (2023-10-01-preview). The Workloads SDK does exist for 2023-04-01 API version.
The breaking change review was already done for the swagger and we had the approval for the same.
https://github.com/Azure/azure-rest-api-specs/pull/25896

**Comment by ronniegeraghty** [comment]
This will be a breaking change for the SDK. Users of the current SDKs will lose features if the API's that map to those methods are moved out of the library in future versions, or users, who update the current SDK expecting the latest API features, won't get them without moving to the new libraries. There isn't a simple migration process for each language to move users who were using the existing libraries to the new libraries and how they'll need to update their code to know which APIs map to the old libraries and which map to the new libraries. 

@ArthurMa1978 since the API breaking change has already been approved how would you like to review the SDK breaking change?

**Comment by ArthurMa1978** [comment]
> Hi @ArthurMa1978
> 
> Could you explain why this would be a breaking change? There is no workloads SDK for this API version (2023-10-01-preview). The Workloads SDK does exist for 2023-04-01 API version. The breaking change review was already done for the swagger and we had the approval for the same. [Azure/azure-rest-api-specs#25896](https://github.com/Azure/azure-rest-api-specs/pull/25896)

The last stable release is API version [2023-04-01](https://github.com/Azure/azure-rest-api-specs/blob/c9a6e0a98a51ebc0c7a346f4fd425ba185f44b31/specification/workloads/resource-manager/readme.md?plain=1#L38) which includes the SapVirtualInstance.

**Comment by ArthurMa1978** [comment]
The PR https://github.com/Azure/azure-rest-api-specs/pull/25896 was approved without the consent of Jeffery or Mike, so this breaking change was not actually got approval.
Even the breaking change was approved, then the swagger should be moved to another folder as it's a separate RP.

**Comment by ajaygit158** [comment]
Hi @ArthurMa1978 

The breaking change was approved by Mike.

![image](https://github.com/Azure/azure-sdk/assets/61898818/c0c32fb5-aff9-43a6-9931-18c00934421e)

Please let us know the way forward to unblock on this.

**Comment by ronniegeraghty** [comment]
> Hi @ArthurMa1978
> 
> The breaking change was approved by Mike.
> 
> ![image](https://private-user-images.githubusercontent.com/61898818/295459449-c0c32fb5-aff9-43a6-9931-18c00934421e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ5MTAxNzUsIm5iZiI6MTcwNDkwOTg3NSwicGF0aCI6Ii82MTg5ODgxOC8yOTU0NTk0NDktYzBjMzJmYjUtYWZmOS00M2E2LTk5MzEtMThjMDA5MzQ0MjFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEwVDE4MDQzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2OWUzNTBiNDAzMzNhZWJkNzA1NGNmMjg1NjVkNWM3MWFlMWYxODlkZTJjZWU2NzQ0MjZmOTI5ODk0MjE1MTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uIJqhsT6bBp3ARFzzsWoUYUeA5TiokF90xv5yrwp8x0)
> 
> Please let us know the way forward to unblock on this.

It looks like quite a few changes were made to the PR after Mike's approval. Did Mike or Jeff know these about the later changes made to the PR?

**Comment by ajaygit158** [comment]
> > Hi @ArthurMa1978
> > The breaking change was approved by Mike.
> > ![image](https://private-user-images.githubusercontent.com/61898818/295459449-c0c32fb5-aff9-43a6-9931-18c00934421e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ5MTAxNzUsIm5iZiI6MTcwNDkwOTg3NSwicGF0aCI6Ii82MTg5ODgxOC8yOTU0NTk0NDktYzBjMzJmYjUtYWZmOS00M2E2LTk5MzEtMThjMDA5MzQ0MjFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEwVDE4MDQzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2OWUzNTBiNDAzMzNhZWJkNzA1NGNmMjg1NjVkNWM3MWFlMWYxODlkZTJjZWU2NzQ0MjZmOTI5ODk0MjE1MTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uIJqhsT6bBp3ARFzzsWoUYUeA5TiokF90xv5yrwp8x0)
> > Please let us know the way forward to unblock on this.
> 
> It looks like quite a few changes were made to the PR after Mike's approval. Did Mike or Jeff know about these later changes made to the PR?

Hi @ronniegeraghty 

The only major change done after Mike's approval was to make the PATCH operation to be asynchronous LRO. The other set of updates were primarily to examples/description for documentation purposes.

https://github.com/Azure/azure-rest-api-specs/pull/25896/files/94bdf1fd11cc4b85db08c016d983f35dc34da881..8c468931477ff18949428890feb924d300d4f8c3#diff-6acd620c8a4c80d9ab811a5be458313b2b85cbca39d1d270e5c2dce681688c7a

There were no changes made post breaking change approval which impacted the folder structure/namespace.

Kindly let us know how we could proceed!

**Comment by ronniegeraghty** [comment]
@ArthurMa1978, how would you like to proceed.

**Comment by ArthurMa1978** [comment]
I'm good for the namespace as it's already approved by Mike.
But please spilt the swagger spec to another folder for SapVirtualInstance, instead of in existing workload folder. Otherwise we can't generate the sdk properly.

**Comment by ronniegeraghty** [comment]
@ArthurMa1978, were good with the 4 level namespaces?

**Comment by weidongxu-microsoft** [comment]
Java also has content of [SAPVirtualInstance.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/workloads/resource-manager/Microsoft.Workloads/stable/2023-04-01/SAPVirtualInstance.json) contained in a GAed azure-resourcemanager-workloads 1.0.0

And "no breaking changes" requirement would require we still put the content of "SAPVirtualInstance.json" in all future "azure-resourcemanager-workloads" releases (even now you got another "azure-resourcemanager-workloads-sapvirtualinstance").

**Comment by ArthurMa1978** [comment]
> @ArthurMa1978 Arthur Ma FTE, we're good with the 4 level namespaces?

Good catch, the namespace shouldn't be 4 level but 3:

- **.NET**: `Azure.ResourceManager.WorkloadsSapVirtualInstance`
- **Java**: `azure-resourcemanager-workloadssapvirtualinstance (com.azure.resourcemanager.workloadssapvirtualinstance)`
- **Go/Golang**: `sdk/resourcemanager/workloadssapvirtualinstance/armworkloadssapvirtualinstance`
- **JavaScript**: `@azure/arm-workloadssapvirtualinstance`
- **Python**: `azure-mgmt-workloadssapvirtualinstance`

**Comment by ArthurMa1978** [comment]
> Java also has content of [SAPVirtualInstance.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/workloads/resource-manager/Microsoft.Workloads/stable/2023-04-01/SAPVirtualInstance.json) released in a GAed azure-resourcemanager-workloads 1.0.0
> 
> And "no breaking changes" requirement would require we still put the content of "SAPVirtualInstance.json" in all future "azure-resourcemanager-workloads" releases (even now you got another "azure-resourcemanager-workloads-sapvirtualinstance"). @JonathanGiles Jonathan Giles FTE @srnagar Srikanta Nagaraja FTE

Right, even the breaking change review was approved but we still need to make backward compatible from sdk side.

**Comment by ronniegeraghty** [comment]
@adityashukla-msft, 
Do you agree to changing the proposed namespaces to what Arthur suggested above: 
- .NET: `Azure.ResourceManager.WorkloadsSapVirtualInstance`
- Java: `azure-resourcemanager-workloadssapvirtualinstance (com.azure.resourcemanager.workloadssapvirtualinstance)`
- Go/Golang: `sdk/resourcemanager/workloadssapvirtualinstance/armworkloadssapvirtualinstance`
- JavaScript: `@azure/arm-workloadssapvirtualinstance`
- Python: `azure-mgmt-workloadssapvirtualinstance`

If so, we can move on to the next phase of the approval process.

**Comment by adityashukla-msft** [comment]
@ronniegeraghty We agree to changing the namespaces. Edited my original comment with the new namespaces.

**Comment by ronniegeraghty** [comment]
We'll now move to the second phase of the review process. The proposed namespaces will be shown to our SDK Architecture team. They will have 1 week to make any objections to the proposed namespaces. If there are no objections by EoB on 2/14 the namespaces will be considered approved, and I'll update this issue with a comment stating so.

**Comment by ronniegeraghty** [comment]
There have been no objections to the namespaces in the description of this issue and the week is up. The namespaces in the description of this issue are considered approved.
